cns/restserver/api.go

```diff
@@ -746,7 +746,7 @@ func (service *HTTPRestService) setOrchestratorType(w http.ResponseWriter, r *ht
 	service.dncPartitionKey = req.DncPartitionKey
 	nodeID = service.state.NodeID
 
-	if nodeID == "" || nodeID == req.NodeID {
+	if nodeID == "" || nodeID == req.NodeID || !service.areNCsPresent() {
 		switch req.OrchestratorType {
 		case cns.ServiceFabric:
```

Comment:
i know you didn't change this, but this `fallthrough` pattern is very unusual in go. if all of these switch tests should actually execute the same case, it's better to do:
```go
case cns.ServiceFabric, cns.Kubernetes, cns.KubernetesCRD, [... other tests ...]:
    // the case
```

Reply:
Good point